gradio/blocks.py

```diff
@@ -783,8 +786,10 @@ def get_block_instance(id: int) -> Block:
             block_config["props"].pop("type", None)
             block_config["props"].pop("name", None)
             style = block_config["props"].pop("style", None)
-            if block_config["props"].get("root_url") is None and root_url:
+            if block_config["props"].get("root_url") is None:
                 block_config["props"]["root_url"] = f"{root_url}/"
+            else:
+                root_urls.add(block_config["props"]["root_url"])
```

Comment:
So this will implicitly add all of the root_urls from the (remote?) block config to the set of trusted URLs for `/proxy=`? Is that okay?

Reply:
Yes, this is necessary. Consider a Gradio app B that `gr.loads()` a Gradio app A from Spaces. Then, suppose Gradio app B is published to Spaces. If a Gradio app C loads the Gradio app B, the files that are generated will actually live on the machine that is running Gradio app A, so app C needs to be able to proxy A.

Reply:
Yeah, figured it'd be something like that. That would probably be good to document in a comment above this line.